### PR TITLE
fix: Fixed import for torch 1.10 release

### PR DESCRIPTION
**Pull request by MateoLostanlen**
Small PR to fix our [unitest](https://github.com/pyronear/pyro-vision/runs/4002060618?check_suite_focus=true)

The function [load_state_dict_from_url is now part of torch.hub](https://github.com/pytorch/pytorch/blob/f4dd88489a77ff9b300bf6f9b34c233ca82f76d7/torch/hub.py#L529)

We need to update the import accordingly

We must therefore set the minimal version of pytorch to 1.10

### CODE REVIEW

**Comment by MateoLostanlen** [comment]
I wonder if setting the minimal version of pytorch to the last release is not too restrictive. @frgfm what do you think ?

It seems you have the same issue in holocron :)

**Comment by frgfm** [comment]
> I wonder if setting the minimal version of pytorch to the last release is not too restrictive. @frgfm what do you think ?
> 
> It seems you have the same issue in holocron :)

Yup I agree, either I update it or do a conditional import by checking the library version :+1: 

Considering `0.10.0` is brand new, either we decide to:
- update it because there are features that we consider critical in this release
- add an upper constraint (<0.10.0)
- make the import conditional

I would go for the latter in this situation :+1:

**Comment by MateoLostanlen** [comment]
It seems to be the best solution indeed, I opened [a PR](https://github.com/frgfm/Holocron/pull/146) on holocron to solve the problem.

**Review by frgfm — CHANGES_REQUESTED** [review]
Actually the function is available in PyTorch since 1.1.0 (https://github.com/pytorch/pytorch/blob/v1.1.0/torch/hub.py#L385), and torchvision was only making a mirror for it up until the last release! So we only need:
```python
from torch.hub import load_state_dict_from_url
```

**Comment by MateoLostanlen** [comment]
Hi @frgfm, thanks for catching this !

**Review by frgfm — DISMISSED** [review]
Thanks for the edits!

**Comment by frgfm** [comment]
About the unittest crash, I'll make a release of Holocron rather soon, we'll need to update the version constraint ;)

**Comment by MateoLostanlen** [comment]
> 
> 
> About the unittest crash, I'll make a release of Holocron rather soon, we'll need to update the version constraint ;)

Ok Let me know when it's done I will wait to go further with this PR and set holocron minimal requirement (0.1.4 I guess).

**Comment by frgfm** [comment]
> > About the unittest crash, I'll make a release of Holocron rather soon, we'll need to update the version constraint ;)
> 
> Ok Let me know when it's done I will wait to go further with this PR and set holocron minimal requirement (0.1.4 I guess).

To avoid waiting for the release, I suggest switching all pylocron requirements ([here](https://github.com/pyronear/pyro-vision/blob/master/requirements.txt#L7) and [here](https://github.com/pyronear/pyro-vision/blob/master/setup.py#L46)) from
```
pylocron>=0.1.3
```
to 
```
-e git+https://github.com/frgfm/Holocron.git#egg=pylocron
```

This will install the latest version from the git master branch instead of the release :+1:

**Comment by MateoLostanlen** [comment]
Hi @frgfm, unfortunately we can' t use `e git+https://github.com/frgfm/Holocron.git#egg=pylocron ` in setup.py. A simple solution while waiting for the new holocron release is to install the requirements and then the package as I do [here](https://github.com/MateoLostanlen/pyro-vision/blob/a48daebad7d35b78123f4cf5cbdeaaf106512258/.github/workflows/builds.yml#L37).

The problem seems to be solved for the moment.

Only one test doesn't pass but it's always the same problem of openfire which has less and less image available. This problem will be solved after the merge of #121

File: .github/workflows/builds.yml
Comment (frgfm) [inline .github/workflows/builds.yml]: That line isn't needed: the next one will install all deps listed in `setup.py` along with the lib. And the latest pypi release of pylocron is a valid one, so I'd suggest to remove this

File: requirements.txt
Comment (frgfm) [inline requirements.txt]: would be safer to go for `pylocron>=0.2.0` and it would guarantee deps consistency between `requirements.txt` and `setup.py`

File: setup.py
Comment (frgfm) [inline setup.py]: `pylocron>=0.2.0`

**Review by frgfm — COMMENTED** [review]
Thanks! I added a few suggestions

**Comment by MateoLostanlen** [comment]
Hi @frgfm, I think you did not understand. I propose here a temporary fix while you make your release because it blocks the pr on vision and engine for the moment. This modification allows to install holocron from the repo without waiting for the release. So you can take your time :)

**Comment by MateoLostanlen** [comment]
ah ok you release 0.2.0, I will make the change then

**Comment by MateoLostanlen** [comment]
all good :)

**Review by frgfm — APPROVED** [review]
All good!

**Pull request merged**
